src/frontegg-mock/src/lib.rs

```diff
@@ -570,6 +617,298 @@ async fn handle_roles_request(State(context): State<Arc<Context>>) -> Json<UserR
+    if let Some(config) = configs.get(&config_id) {
+        config
+            .groups
+            .iter()
+            .find(|g| g.id == group_id)
+            .cloned()
+            .map(Json)
+            .ok_or(StatusCode::NOT_FOUND)
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+async fn handle_update_group_mapping(
+    State(context): State<Arc<Context>>,
+    Path((config_id, group_id)): Path<(String, String)>,
+    Json(mut updated_group): Json<GroupMapping>,
+) -> Result<Json<GroupMapping>, StatusCode> {
+    let mut configs = context.sso_configs.lock().unwrap();
+    if let Some(config) = configs.get_mut(&config_id) {
+        if let Some(group) = config.groups.iter_mut().find(|g| g.id == group_id) {
+            updated_group.id.clone_from(&group.id);
+            updated_group.sso_config_id.clone_from(&group.sso_config_id);
+            *group = updated_group.clone();
+            Ok(Json(updated_group))
+        } else {
+            Err(StatusCode::NOT_FOUND)
+        }
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+async fn handle_delete_group_mapping(
+    State(context): State<Arc<Context>>,
+    Path((config_id, group_id)): Path<(String, String)>,
+) -> StatusCode {
+    let mut configs = context.sso_configs.lock().unwrap();
+    if let Some(config) = configs.get_mut(&config_id) {
+        let initial_len = config.groups.len();
+        config.groups.retain(|g| g.id != group_id);
+        if config.groups.len() < initial_len {
+            StatusCode::OK
+        } else {
+            StatusCode::NOT_FOUND
+        }
+    } else {
+        StatusCode::NOT_FOUND
+    }
+}
+
+async fn handle_get_default_roles(
+    State(context): State<Arc<Context>>,
+    Path(config_id): Path<String>,
+) -> Result<Json<DefaultRoles>, StatusCode> {
+    let configs = context.sso_configs.lock().unwrap();
+    if let Some(config) = configs.get(&config_id) {
+        Ok(Json(config.default_roles.clone()))
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+async fn handle_set_default_roles(
+    State(context): State<Arc<Context>>,
+    Path(config_id): Path<String>,
+    Json(default_roles): Json<DefaultRoles>,
+) -> Result<(StatusCode, Json<DefaultRoles>), StatusCode> {
+    let mut configs = context.sso_configs.lock().unwrap();
+    if let Some(config) = configs.get_mut(&config_id) {
+        config.default_roles = default_roles.clone();
+        config.role_ids.clone_from(&config.default_roles.role_ids);
+        Ok((StatusCode::CREATED, Json(default_roles)))
+    } else {
+        Err(StatusCode::NOT_FOUND)
+    }
+}
+
+async fn handle_clear_default_roles(
+    State(context): State<Arc<Context>>,
+    Path(config_id): Path<String>,
+) -> Result<Json<DefaultRoles>, StatusCode> {
+    let mut configs = context.sso_configs.lock().unwrap();
+    if let Some(config) = configs.get_mut(&config_id) {
+        config.default_roles.role_ids.clear();
+        config.role_ids.clear();
```

Comment:
Is there documentation for this one? I can't find it. If this is something we made that isn't part of normal frontegg, I'm not sure why we'd need this, since `handle_set_default_roles` is a `PUT`.

We seem to be clearing all role_ids, not just the default ones, which seems surprising. I have no idea if this is correct, since I can't find docs for this.

Reply:
Hm yes this is probably an overlook on my part, will check this out!

Yep it is part of the Frontegg API, here is the link to the documentation:

> https://docs.frontegg.com/reference/ssorolescontrollerv1_setssodefaultroles

Reply:
That's a link to the docs for the `PUT`, which we handle with `handle_set_default_roles`. I don't see any docs for the `DELETE`, which seems to be what we're trying to handle here.

Reply:
Ah yes! Indeed, this endpoint does not really exist and we actually never used it, it must have been something that I had in the old mock service that I've used for local testing only.